src/main/java/org/mule/api/annotation/NoExtend.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.api.annotation;
+
+import static java.lang.annotation.ElementType.TYPE;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Target;
+
+/**
+ * Indicates that a class must not be extended by clients, as it is designed to be extended internally on a module.
```

Comment:
", as it is designed to be extended internally on a module" -> I would remove this since it may not make sense in every context. I would just say that is not meant to be extend and that's it.